### ISSUE TITLE
Undeterministic Output

### ISSUE THREAD

**Issue by lucasavila00** [tommywalkie/excalidraw-cli#6]
**Describe the bug**

I have some .excalidraw files in a github project to document something.

I use the binary like this: `excalidraw-cli docs/readme docs/readme` and it goes through all the files in the folder and outputs them as expected.

However every time I run this CLI it will give me different output files even though I haven't changed input files (I haven't been able to **SEE** 👁️  the differences but git will mark them as changed).

It's just annoying (as in not breaking my flow) because if I change one file I must submit a commit where it seems all files changed.

**Expected behavior**

The same input .excalidraw file should produce the **exact same** PNG, don't matter where/when I run it.

**Excalidraw input file**

This is a sample project where I created a PR where I changed just 1 of 2 excalidraw files.
After building the project with excalidraw-cli and pushing the change the PR has changes to the 2 PNG files.

https://github.com/lucasavila00/excalidraw-non-deterministic/pull/1/files

**Environment (please complete the following information):**
- OS version: Ubuntu 20.04 LTS
- Node version: v12.18.0

**Comment by tommywalkie** [comment]
Can confirm this odd behaviour is happening to me, using your repo as reference.

![image](https://user-images.githubusercontent.com/16148332/86275678-eb654e80-bbd3-11ea-85dd-f5719eadf2f0.png)

Will proceed by looking if by any chance this is a `node-canvas` npm package related issue.
Before saving the file, the stream is generated with this method :

https://github.com/tommywalkie/excalidraw-cli/blob/bb422ccb0712ed74eab09e32e6db26c11ebf9055/src/compute.js#L18

**Comment by tommywalkie** [comment]
Not necessarly related but might be worth checking the stream object info if there is a `Date` object. Reference : https://github.com/matplotlib/matplotlib/issues/6317/#issuecomment-211665983

**Comment by tommywalkie** [comment]
@lucasavila00 Bingo ! Just forked your repo ([link](https://github.com/tommywalkie/excalidraw-non-deterministic)) and switched a few settings in `excalidraw-cli` so it makes **transparent** background instead of the one stored in `json.appState.viewBackgroundColor` (usually, it's `white`). 

```diff
const ctx = canvas.getContext("2d") 
 rc.rectangle( 
     0, 
     0, 
     maxDimensions[0], 
     maxDimensions[1], 
-     { fill: json.appState.viewBackgroundColor, fillStyle: 'solid', stroke: 'transparent' } 
+     { fill: 'transparent', fillStyle: 'solid', stroke: 'transparent' } 
 )
```

When trying to generate new PNGs with the mentionned settings after https://github.com/tommywalkie/excalidraw-non-deterministic/commit/8aa71f7489d5571086aed8b1e591be22e2f258ca commit, `git` **doesn't detect any change**.

**Comment by tommywalkie** [comment]
@lucasavila00 Sorry for the delay, I could find **a fix**.
So I just reverted back the PNGs to the way there were before the fork (with a white background) =>  https://github.com/tommywalkie/excalidraw-non-deterministic/commit/9f61c9a9842c7f90259194bb67932fadc9ead3f0
And then I tried generating PNGs with the following setting which makes use of the `roughness` attribute with value 0 for perfect rectangles, and `git` doesn't detect changes.
```js
const ctx = canvas.getContext("2d")
rc.rectangle( 0, 0, maxDimensions[0], maxDimensions[1], { 
     fill: json.appState.viewBackgroundColor,
     fillStyle: 'solid',
     stroke: json.appState.viewBackgroundColor,
     roughness: 0
})
```
Can add the fix right now and it should be available in `0.3.7` 👍

**Comment by tommywalkie** [comment]
Fix is available as of `0.3.8` 📣

**Comment by lucasavila00** [comment]
@tommywalkie  You're awesome, the best!!!

Thanks :)
